### ISSUE TITLE
New `wasabi` sink

### ISSUE THREAD

**Issue by zsherman** [timberio/vector#1782]
[Wasabi](https://wasabi.com/) offers a very fast, cost-effective storage alternative to s3 that some customers are interested in sending data to. It should be fully compatible with the s3 protocols, so I'm hoping this shouldn't be too much additional work.

**Comment by binarylogic** [comment]
@LucioFranco with your recent changes we should be able to wrap the `aws_s3` sink, right?

**Comment by LucioFranco** [comment]
@binarylogic yeah, looks like directly wrapping s3 should be fine though at that rate it just seems like we should tell users to use the aws_s3 sink since they advertise as an s3 alternative?
